File: 백준/Gold/1253. 좋다/좋다.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int A[] = new int[N];
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i = 0; i < N; i++){
            A[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(A);
        
        int count = 0;
        for(int k = 0; k < N; k++){
            long find = A[k];
            int i = 0; //A 배열의 시작
            int j = N-1; // 배열의 끝
            
            while(i < j){
                if(A[i]+ A[j] == find){
                    if(i!=k && j !=k){
                        count++;
                        break;
                    }
                    else if(i==k){
                        i++;
                    }
                    else if(j==k){
                        j--;
                    }
                }
                else{
                    if(A[i] + A[j] < find)
                    {
                        i++;
                    }
                    else{
                        j--;
                    }
                }
            }
        }
        System.out.println(count);
  
    }
}

File: 백준/Gold/1253. 좋다/좋다.py
import sys
input = sys.stdin.readline

N = int(input())
A = list(map(int, input().split()))

A.sort()
count = 0
for k in range(N):
    start = 0
    end = N-1
    find = A[k] # 찾아야할 값
    while start < end:
        if A[start] + A[end] == find:
            if start != k and end != k:
                count +=1
                break
            elif start == k:
                start +=1
            else:
                end -=1
        elif A[start] + A[end] < find:
            start +=1
        else:
            end -=1
print(count)
        